client/src/containers/frontend/PasswordReset.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component, PropTypes } from 'react';
+import { passwordsAPI } from 'api';
+import { Form } from 'components/global';
+import { entityStoreActions, notificationActions, currentUserActions } from 'actions';
+import { browserHistory } from 'react-router';
+import { get } from 'lodash';
+import { select } from 'utils/entityUtils';
+import { connect } from 'react-redux';
+const { request, flush } = entityStoreActions;
+
+class PasswordReset extends Component {
+
+  static propTypes = {};
+
+  constructor() {
+    super();
+    this.state = {
+      password: '',
+      passwordConfirmation: ''
+    };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  static mapStateToProps(state, ownProps) {
+    return {
+      response: get(state.entityStore.responses, 'reset-password'),
+      users: get(state.entityStore.entities, 'users')
+    };
+  }
+
+  componentWillUnmount() {
+    this.props.dispatch(flush(['reset-password']));
+  }
+
+  handleSubmit(event) {
+    event.preventDefault(event.target);
+    this.props.dispatch(request(passwordsAPI.update(this.state.password, this.state.passwordConfirmation, this.props.params.resetToken), 'reset-password')).promise.then((response) => {
+      this.postUpdate(response.data);
```

Comment:
Is this where the reducers come into play? I didn't make/update anything in the reducers, so I'm guessing this could be further refined to not directly rely on the response.